### PR TITLE
currency: Add methods to pairs type

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Adds a method to see if a code is contained in the currency pairs list. `Has()`
Adds a method to construct from a pairs list with a list of currencies. `GetPairsByCurrencies`

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: currency/pairs.go [hunk 1]
```diff
@@ -159,6 +170,19 @@ func (p Pairs) GetPairsByFilter(filter Code) Pairs {
 	return pairs
 }
 
+// GetPairsByCurrencies returns all pairs that have both matches to the
+// currencies passed in. This allows for the construction of enabled pairs by
+// required currency codes.
+func (p Pairs) GetPairsByCurrencies(enabled Currencies) Pairs {
```
Comment (gloriousCode) [inline currency/pairs.go:176]: `enabled` here bears no meaning, passing in a `Currencies` does not mean that they are enabled. Please rn to a more descriptive variable such as `currencies`

File: currency/pairs_test.go [hunk 2]
```diff
@@ -505,3 +505,61 @@ func BenchmarkRemovePairsByFilter(b *testing.B) {
...
+	if !pairs.Has(XRP) {
+		t.Fatalf("expected %s to be %v", XRP, true)
+	}
+	if pairs.Has(ATOM3L) {
+		t.Fatalf("expected %s to be %v", ATOM3L, false)
+	}
+}
+
+func TestGetPairsByCurrencies(t *testing.T) {
+	available := Pairs{
```
Comment (gloriousCode) [inline currency/pairs_test.go:541]: parallel

File: currency/pairs_test.go [hunk 3]
```diff
@@ -505,3 +505,61 @@ func BenchmarkRemovePairsByFilter(b *testing.B) {
 		_ = pairs.RemovePairsByFilter(USD)
 	}
 }
+
+func TestHas(t *testing.T) {
+	pairs := Pairs{
```
Comment (gloriousCode) [inline currency/pairs_test.go:510]: parallel

**Review by gloriousCode — CHANGES_REQUESTED** [review]
🎉 some nice little helpers

File: currency/pairs_test.go [hunk 4]
```diff
@@ -505,3 +505,61 @@ func BenchmarkRemovePairsByFilter(b *testing.B) {
...
+		NewPair(DAI, XRP),
+	}
+
+	enabled := available.GetPairsByCurrencies(Currencies{USD})
+	if len(enabled) != 0 {
+		t.Fatalf("received %v but expected %v", enabled, "no pairs")
+	}
+
+	enabled = available.GetPairsByCurrencies(Currencies{USD, BTC})
+	if !enabled.Contains(NewPair(USD, BTC), false) {
```
[the remark this answers is not in the document]
Comment (shazbert) [reply currency/pairs_test.go:557]: It was just a copy-paste situation. I will change to be more specific.

File: currency/pairs.go [hunk 5]
```diff
@@ -133,6 +133,17 @@ func (p Pairs) Contains(check Pair, exact bool) bool {
 	return false
 }
 
+// Has checks to see if a specified currency code exists inside a currency pair
+// array
+func (p Pairs) Has(check Code) bool {
```
[the remark this answers is not in the document]
Comment (shazbert) [reply currency/pairs.go:138]: 😭

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes

**Review by thrasher- — APPROVED** [review]
utACK 👍

**Pull request merged**